utilities/backupable/backupable_db.cc

```diff
@@ -281,14 +298,16 @@ class BackupEngineImpl : public BackupEngine {
     return GetSharedChecksumDirRel() + "/" + (tmp ? "." : "") + file +
            (tmp ? ".tmp" : "");
   }
-  inline std::string GetSharedFileWithChecksum(const std::string& file,
-                                               const uint32_t checksum_value,
-                                               const uint64_t file_size) const {
+  inline std::string GetSharedFileWithChecksum(
+      const std::string& file, const uint32_t checksum_value,
+      const uint64_t file_size, const std::string& db_session_id) const {
     assert(file.size() == 0 || file[0] != '/');
     std::string file_copy = file;
-    return file_copy.insert(file_copy.find_last_of('.'),
-                            "_" + ROCKSDB_NAMESPACE::ToString(checksum_value) +
-                                "_" + ROCKSDB_NAMESPACE::ToString(file_size));
+    const std::string suffix =
+        UseNewNaming() ? db_session_id : ROCKSDB_NAMESPACE::ToString(file_size);
```

Comment:
Just because DBImpl session IDs are safe to use in file names does not mean that will hold for all DB implementations. We need to do something safe.

Reply:
Right, also same as the above. Will fall back to file size when the id is empty.